java-frontend/src/main/java/org/sonar/java/ast/JavaAstScanner.java

```diff
@@ -131,19 +140,13 @@ public static void scanSingleFileForTests(File file, VisitorsBridge visitorsBrid
     if (!file.isFile()) {
       throw new IllegalArgumentException("File '" + file + "' not found.");
     }
-    JavaAstScanner scanner = create(conf, visitorsBridge);
-
-    scanner.scan(Collections.singleton(file));
-  }
-
-  private static JavaAstScanner create(JavaConfiguration conf, @Nullable VisitorsBridge visitorsBridge) {
-    JavaAstScanner astScanner = new JavaAstScanner(JavaParser.createParser(conf.getCharset()));
+    JavaAstScanner astScanner = new JavaAstScanner(JavaParser.createParser(conf.getCharset()), null);
     if (visitorsBridge != null) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Change this condition so that it does not always evaluate to "true" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2583)